examples/seq2seq/seq2seq_trainer.py

```diff
@@ -114,23 +123,31 @@ def _get_train_sampler(self) -> Optional[torch.utils.data.sampler.Sampler]:
                 else DistributedSampler(self.train_dataset)
             )
 
-    def compute_loss(self, model, inputs):
-        labels = inputs.pop("labels")
-        outputs = model(**inputs, use_cache=False)
-        logits = outputs[0]
-        return self._compute_loss(logits, labels)
-
-    def _compute_loss(self, logits, labels):
+    def _compute_loss(self, model, inputs):
         if self.args.label_smoothing == 0:
-            # Same behavior as modeling_bart.py
-            loss_fct = torch.nn.CrossEntropyLoss(ignore_index=self.config.pad_token_id)
-            assert logits.shape[-1] == self.vocab_size
-            loss = loss_fct(logits.view(-1, logits.shape[-1]), labels.view(-1))
+            # set all ids to -100 to be ignored
+            if self.data_args is not None and self.data_args.ignore_pad_token_for_loss:
+                assert self.config.pad_token_id >= 0, "Make sure that `config.pad_token_id` is correcly defined"
+                inputs["labels"][inputs["labels"] == self.config.pad_token_id] = -100
```

Comment:
This keeps 100% backwards compatibility

Reply:
Will this cause a TPU issue @LysandreJik ?

Reply:
Why would this cause a TPU issue? All of our models work with -100 to ignore CE loss

Reply:
I think some tensor manipulations/assignments on TPU requiring sending the tensor back to CPU to do the op then returning it to CPU. Lys told me it was bad to `assert -100 in inputs['labels']`, for that reason. In that case we could do this in the collator I guess.